src/components/designSystem/Icon/Icon.tsx

```diff
@@ -24,49 +22,46 @@ enum IconAnimationEnum {
   spin = 'spin',
   pulse = 'pulse',
 }
+enum IconSizeEnum {
+  small = '12px',
+  medium = '16px',
+  large = '24px',
+}
 interface IconProps {
   name: IconName
-  size?: 'small' | 'medium' | 'large'
+  size?: keyof typeof IconSizeEnum
   color?: IconColor
   className?: string
   animation?: keyof typeof IconAnimationEnum
   onClick?: () => {} | void | Promise<void>
 }
```

Comment:
Instead creating new types, you can directly extract types from cva like this:
```suggestion
import { cva, VariantProps } from 'class-variance-authority'

export type IconVariantProps = VariantProps<typeof iconStyles>;

interface IconProps extends IconVariantProps {
  name: IconName
  className?: string
  onClick?: () => {} | void | Promise<void>
}
```

See https://cva.style/docs/getting-started/typescript

Reply:
Sick! love it, updated the code